src/analysis/processing/qgsalgorithmlinedensity.cpp

```diff
@@ -54,6 +54,10 @@ void QgsLineDensityAlgorithm::initAlgorithm( const QVariantMap & )
   addParameter( new QgsProcessingParameterDistance( QStringLiteral( "RADIUS" ), QObject::tr( "Search radius" ), 10, QStringLiteral( "INPUT" ), false, 0 ) );
   addParameter( new QgsProcessingParameterDistance( QStringLiteral( "PIXEL_SIZE" ), QObject::tr( "Pixel size" ), 10, QStringLiteral( "INPUT" ), false ) );
 
+  std::unique_ptr< QgsProcessingParameterString > createOptsParam = qgis::make_unique< QgsProcessingParameterString >( QStringLiteral( "CREATE_OPTIONS" ), QObject::tr( "Creation options" ), QVariant(), false, true );
```

Comment:
What do you think about moving this to a new "addStandardParameter" method in QgsProcessingAlgorithm which takes a "StandardParameter" enum value? It'd avoid all this duplicate code here (and likely in similar other places)

Reply:
You mean having a set of widely used parameters in the base class, so the subclasses can add them with the single line of code? Sounds like good idea.

Reply:
Exactly!

Reply:
Maybe we should pass a VariantMap instead of an enum to allow configuring parameter's flags/metadata?

Reply:
I'd still use an enum for the parameter "preset" eg "RasterCreationOptions", "VectorCreationOptions", but a variant map "config" argument could be useful...